.github/workflows/master-push.yml

```diff
@@ -0,0 +1,564 @@
+              package-${{ matrix.platform }}-${{ matrix.xcode-version }}-RealmSwift-release.zip
+            key: ${{ matrix.platform }}-${{ matrix.xcode-version }}-RealmSwift-release
+        - name: Restore package Realm release for simulator
+          uses: actions/cache/restore@v3
+          if: ${{ matrix.platform == 'ios' }} || ${{ matrix.platform == 'tvos' }} || ${{ matrix.platform == 'watchos' }} || ${{ matrix.platform == 'visionos' }}
+          with:
+            path: |
+              package-${{ matrix.platform }}simulator-${{ matrix.xcode-version }}-Realm-release.zip
+            key: ${{ matrix.platform }}simulator-${{ matrix.xcode-version }}-Realm-release
+        - name: Restore package RealmSwift release for simulator
+          uses: actions/cache/restore@v3
+          if: ${{ matrix.platform == 'ios' }} || ${{ matrix.platform == 'tvos' }} || ${{ matrix.platform == 'watchos' }} || ${{ matrix.platform == 'visionos' }}
+          with:
+            path: |
+              package-${{ matrix.platform }}simulator-${{ matrix.xcode-version }}-RealmSwift-release.zip
+            key: ${{ matrix.platform }}simulator-${{ matrix.xcode-version }}-RealmSwift-release
+        - name: Restore package Realm static
+          uses: actions/cache/restore@v3
+          if: ${{ matrix.platform == 'ios' }}
+          with:
+            path: |
+              package-${{ matrix.platform }}-${{ matrix.xcode-version }}-Realm-static.zip
+            key: ${{ matrix.platform }}-${{ matrix.xcode-version }}-Realm-static
+        - name: Restore package Realm static for simulator
+          uses: actions/cache/restore@v3
+          if: ${{ matrix.platform == 'ios' }}
+          with:
+            path: |
+              package-${{ matrix.platform }}simulator-${{ matrix.xcode-version }}-Realm-static.zip
+            key: ${{ matrix.platform }}simulator-${{ matrix.xcode-version }}-Realm-static
+        - name: Create xcframework
+          run: |
+            sh -x build.sh release-create-xcframework_${{ matrix.xcode-version }} ${{ matrix.platform }}
+        - name: Cache xcframework product
+          uses: actions/cache/save@v3
+          with:
+            path: |
+              realm-${{ matrix.platform }}-${{ matrix.xcode-version }}.zip
+            key: realm-${{ matrix.platform }}-${{ matrix.xcode-version }}
+            enableCrossOsArchive: true
+  create-package-version: # Cretes a release package for each xcode version
+      runs-on: macos-13
+      name: Package release for each xcode version
+      needs: [package-xcframework-platform, prepare]
+      strategy:
+        matrix:
+          xcode-version: ${{ fromJSON(needs.prepare.outputs.XCODE_VERSIONS_MATRIX) }}
+      steps:
+        - name: Checkout
+          uses: actions/checkout@v4
+        - uses: maxim-lobanov/setup-xcode@v1
+          with:
+            xcode-version: ${{ env.XCFRAMEWORK_VERSION }} 
+        - name: Restore ios xcframework
+          uses: actions/cache/restore@v3
```

Comment:
Why is this using caches instead of artifacts to pass files between jobs? It seems like there's a lot of room for things to go very weird due to it not being at all what caches are intended for.